### PR TITLE
Fix serverOnly mode not working due to registering the new menu type

### PR DESCRIPTION
**Pull request by ChiefArug**
### Description 
Fixes [KubeJS 2001.6.3-build50 | serverOnly=true | Still requires me to have kubejs on client (BC1A0B2D)](https://kubejs.com/support/tickets/BC1A0B2D), caused by KubeJS' menutype always registering, even if serverOnly is enabled.

#### Other details

### CODE REVIEW

File: common/src/main/java/dev/latvian/mods/kubejs/gui/KubeJSMenu.java [hunk 1]
```diff
@@ -15,10 +16,12 @@
 
 public class KubeJSMenu extends AbstractContainerMenu {
 	public static final DeferredRegister<MenuType<?>> MENU_TYPES = DeferredRegister.create("kubejs", Registries.MENU);
-	public static final Supplier<MenuType<KubeJSMenu>> KUBEJS_MENU = MENU_TYPES.register("menu", () -> MiscPlatformHelper.get().createMenuType());
```
Comment (MaxNeedsSnacks) [inline common/src/main/java/dev/latvian/mods/kubejs/gui/KubeJSMenu.java:18]: Why did you remove `final` here if you don't mind me asking?

Comment (ChiefArug) [reply common/src/main/java/dev/latvian/mods/kubejs/gui/KubeJSMenu.java:18]: I was planning on also only filling this variable if serverOnly mode was off, like is done for some other registry KubeJS has, but then realised I didn't need to do that and forgot to revert it.
